test/test_webservice.py

```diff
@@ -497,3 +526,74 @@ def test_set_retries_not_reached(self):
         self.assertTrue(TEMP_ERRORS_RETRIES > 1)
         self.assertEqual(request.mark_for_retry(), 1)
         self.assertFalse(request.max_retries_reached())
+
+    def test_queryargs(self):
+        request = WSRequest(
+            url='http://example.org/path?a=1',
+            method='GET',
+            handler=dummy_handler,
+            queryargs={'a': 2, 'b': 'x%20x', 'c': '1+2', 'd': '&', 'e': '?'},
+        )
+        expected = 'http://example.org/path?a=1&a=2&b=x x&c=1+2&d=%26&e=?'
```

Comment:
@phw I wonder about expected behavior here when url has already a query (`?a=1`), the current code just append `queryargs` key/value elements.
I don't think that's a case happening in current code, but it would probably make more sense to actually replace existing values with those passed from `queryargs`.

Reply:
I tend to agree that I would have expected replacement. On the other hand there could be even a use case for the behavior: If you access an API where some kind of access token gets transferred by query parameter you could define the base URL as `https://example.com/api/?token=12345`, and then still do extra requests with additional query parameters.

Currently this is all hypothetical though, as this is not used in practice.

In the end I have no strong opinion either way.

Reply:
I'll keep it as it is now, we can always change it later if needed